packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -158,7 +158,10 @@ class _LayoutBuilderElement<ConstraintType extends Constraints> extends RenderOb
 
   @override
   void markNeedsBuild() {
-    super.markNeedsBuild();
+    // Calling super.markNeedsBuild is not needed. This Element does not need
```

Comment:
If this is the case, will this class's performRebuild be called at all? if the assumption is this element is never dirty, we should probably assert the performRebuild not getting called at all

Reply:
Yes, when the parent Element calls `updateChild` (i.e. in treewalk).

Reply:
ah I see, I just found out the `RenderObjectElement` overrides the `update`